integration_test.go

```diff
@@ -1133,3 +1134,46 @@ func TestGRPCReflectionSource(t *testing.T) {
 		})
 	}
 }
+
+type protoReflectService struct {
+	sleepDuration time.Duration // If non-zero, handler sleeps for the given duration.
+}
+
+func (p protoReflectService) ServerReflectionInfo(s rpb.ServerReflection_ServerReflectionInfoServer) error {
+	if p.sleepDuration != 0 {
+		// Sleep until the given duration or until stream context completes.
+		select {
+		case <-s.Context().Done():
+		case <-time.After(p.sleepDuration):
+		}
+	}
+	return nil
+}
+
+func TestGPCReflectionTimeout(t *testing.T) {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+
+	// Sleep duration must be longer than go test timeout (30s) to
+	// catch client which does not respect timeouts in reflection calls.
+	// See MakeFile(test_ci,test) for go test timeout.
+	reflectSvc := protoReflectService{sleepDuration: time.Second * 31}
```

Comment:
Instead of relying on the test timeout (which may change), we can instead have the `protoReflectService` block on a context until the test is complete. So instead of a `sleepDuration`, we can pass a channel or a context.

Reply:
Thanks! moved this to use a channel 👍